File: driver-service/src/main/java/com/dulfinne/taxi/driverservice/service/DriverService.java
package com.dulfinne.taxi.driverservice.service;

import com.dulfinne.taxi.driverservice.dto.request.DriverRequest;
import com.dulfinne.taxi.driverservice.dto.request.PointRequest;
import com.dulfinne.taxi.driverservice.dto.response.DriverResponse;
import com.dulfinne.taxi.driverservice.dto.response.PointResponse;
import org.springframework.data.domain.Page;

public interface DriverService {

  Page<DriverResponse> getAllDrivers(Integer offset, Integer limit, String sortField);

  DriverResponse getDriverByUsername(String username);

  DriverResponse saveDriver(String username, DriverRequest driverRequest);

  DriverResponse updateDriver(String username, DriverRequest driverRequest);

  void deleteDriver(String username);

  DriverResponse assignCarToDriver(String username, Long carId);

  DriverResponse removeCarFromDriver(String username);

  PointResponse getDriverLocation(String username);

  PointResponse updateDriverLocation(String username, PointRequest request);
}

File: ride-service/src/main/kotlin/com/dulfinne/taxi/rideservice/service/DriverService.kt
package com.dulfinne.taxi.rideservice.service

import com.dulfinne.taxi.rideservice.dto.request.RatingRequest
import com.dulfinne.taxi.rideservice.dto.response.AvailableRideResponse
import com.dulfinne.taxi.rideservice.dto.response.RideResponse
import org.springframework.data.domain.Page

interface DriverService {
    fun getAvailableRides(username: String, radius: Int): List<AvailableRideResponse>
    fun acceptRide(rideId: Long, driverUsername: String): RideResponse
    fun startRide(rideId: Long, driverUsername: String): RideResponse
    fun finishRide(rideId: Long, driverUsername: String): RideResponse
    fun ratePassenger(rideId: Long, driverUsername: String, request: RatingRequest)
    fun getAllDriverRides(driverUsername: String, offset: Int, limit: Int, sortField: String): Page<RideResponse>
    fun getRideById(driverUsername: String, rideId: Long): RideResponse
}